packages/builtin/string.pony

```diff
@@ -41,6 +41,14 @@ class val String is (Seq[U8] & Comparable[String box] & Stringable)
       _set(_size, 0)
     end
 
+  new iso from_iso_array(data: Array[U8] iso) =>
+    """
+    Create a string from an array, reusing the underlying data pointer
+    """
+    _size = data.size()
+    _alloc = data.space()
+    _ptr = (consume data)._cstring()._unsafe()._offset(0)
```

Comment:
Is the `_offset` call really necessary?  As far as I can tell `_offset(0)` would essentially be a no-op that returns the same pointer you started with.